src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3299,6 +3301,343 @@ tx_gateway_frontend::get_all_transactions() {
+    } else if (tx.status == tm_transaction::tx_status::ready) {
+        auto r1 = co_await fetch_tx(tx.id, tx.etag);
+        if (!r1) {
+            vlog(
+              txlog.warn,
+              "Fetching a cached tx from old leader (tx:{} pid:{} etag:{}"
+              "tx_seq:{}) failed with error:{}",
+              tx.id,
+              tx.pid,
+              tx.etag,
+              tx.tx_seq,
+              r1.error());
+            // Return possibly stale data
+            co_return tx;
+        }
+
+        old_tx = r1.value();
+
+        if (old_tx.status == tm_transaction::tx_status::ongoing) {
+            if (old_tx.pid != tx.pid) {
+                vlog(
+                  txlog.warn,
+                  "An ongoing tx (tx:{} pid:{} etag:{} tx_seq:{}) should"
+                  "have the same pid as a preceding ready tx (pid:{} "
+                  "etag:{} tx_seq:{})",
+                  old_tx.id,
+                  old_tx.pid,
+                  old_tx.etag,
+                  old_tx.tx_seq,
+                  tx.pid,
+                  tx.etag,
+                  tx.tx_seq);
+                co_return tx_errc::unknown_server_error;
+            }
+        } else if (old_tx.status == tm_transaction::tx_status::ready) {
+            if (old_tx.tx_seq > tx.tx_seq) {
+                vlog(
+                  txlog.warn,
+                  "A preceding ready tx (tx:{} pid:{} etag:{} tx_seq:{})"
+                  "should have lesser tx_seq than current ready tx (pid:{} "
+                  "etag:{} tx_seq:{})",
+                  old_tx.id,
+                  old_tx.pid,
+                  old_tx.etag,
+                  old_tx.tx_seq,
+                  tx.pid,
+                  tx.etag,
+                  tx.tx_seq);
+                co_return tx_errc::unknown_server_error;
+            }
+            old_tx = tx;
+        } else if (old_tx.tx_seq < tx.tx_seq) {
+            old_tx = tx;
+        } else {
+            vlog(
+              txlog.warn,
+              "cached tx (tx:{} pid:{} etag:{} tx_seq:{} status:{})"
+              "isn't alighted with persisted ready tx (pid:{} etag:{} "
+              "tx_seq:{})",
+              old_tx.id,
+              old_tx.pid,
+              old_tx.etag,
+              old_tx.tx_seq,
+              tx.pid,
+              tx.etag,
+              tx.tx_seq);
+            co_return tx_errc::unknown_server_error;
+        }
+    } else if (
+      tx.status == tm_transaction::tx_status::prepared
+      || tx.status == tm_transaction::tx_status::aborting) {
+        auto r1 = co_await fetch_tx(tx.id, tx.etag);
+        if (!r1) {
+            vlog(
+              txlog.warn,
+              "Fetching a cached tx from old leader (tx:{} pid:{} etag:{}"
+              "tx_seq:{}) failed with error:{}",
+              tx.id,
+              tx.pid,
+              tx.etag,
+              tx.tx_seq,
+              r1.error());
+            // Return possibly stale data
+            co_return tx;
```

Comment:
Should we really return stale data ? Given we have no coordinator epoch available in the response client will not be able to determine if the result is stale or not

Reply:
cc @rystsov 

I think it is ok. It still raw api a little bit, so it is not a big problem?